packages/arb-rpc-node/batcher/sequencerBatcher.go

```diff
@@ -202,6 +221,93 @@ func (b *SequencerBatcher) SendTransaction(_ context.Context, startTx *types.Tra
 	l2Message := message.NewSafeL2Message(batch)
 	seqMsg := message.NewInboxMessage(l2Message, b.sequencer, new(big.Int).Set(msgCount), big.NewInt(0), b.latestChainTime.Clone())
 
+	logCount, err := b.db.GetLogCount()
+	if err != nil {
+		return err
+	}
+
+	txBatchItem := inbox.NewSequencerItem(totalDelayedCount, seqMsg, prevAcc)
+	err = core.DeliverMessagesAndWait(b.db, prevAcc, []inbox.SequencerBatchItem{txBatchItem}, []inbox.DelayedMessage{}, nil)
+	if err != nil {
+		return err
+	}
+
+	var sequencedTxs []*types.Transaction
+	var sequencedBatchItems []inbox.SequencerBatchItem
+	var eachSucceeded []bool
+
+	newLogCount, err := b.db.GetLogCount()
+	if err != nil {
+		return err
+	}
+	txLogs, err := b.db.GetLogs(logCount, new(big.Int).Sub(newLogCount, logCount))
+	if err != nil {
+		return err
+	}
```

Comment:
The machine may not be done processing the messages here. I think we might need to wait for it to be idle?

Reply:
I thought DeliverMessagesAndWait did that.. oops. We do need to wait for the machine to be idle.